cli/src/main.rs

```diff
@@ -0,0 +1,375 @@
+#[derive(Clap, Debug)]
+struct Opts {
+    /// Address of the Multisig program.
+    #[clap(long)]
+    multisig_program_id: Pubkey,
+
+    /// The keypair to sign and pay with. Defaults to ~/.config/solana/id.json.
+    #[clap(long)]
+    keypair_path: Option<PathBuf>,
+
+    #[clap(subcommand)]
+    subcommand: SubCommand
+}
+
+#[derive(Clap, Debug)]
+enum SubCommand {
+    /// Create a new multisig address.
+    CreateMultisig(CreateMultisigOpts),
+
+    /// Show the owners and threshold of the given multisig.
+    ShowMultisig(ShowMultisigOpts),
+
+    /// Show the details of a transaction.
+    ShowTransaction(ShowTransactionOpts),
+
+    /// Propose replacing a program with that in the given buffer account.
+    ProposeUpgrade(ProposeUpgradeOpts),
+
+    /// Approve a proposed transaction.
+    Approve(ApproveOpts),
+}
+
+#[derive(Clap, Debug)]
+struct CreateMultisigOpts {
+    /// How many signatures are needed to approve a transaction.
+    #[clap(long)]
+    threshold: u64,
+
+    /// The public keys of the multisig owners, who can sign transactions.
+    #[clap(long = "owner")]
+    owners: Vec<Pubkey>,
+}
+
+#[derive(Clap, Debug)]
+struct ProposeUpgradeOpts {
+    /// The multisig account whose owners should vote for this proposal.
+    #[clap(long)]
+    multisig_address: Pubkey,
+
+    /// The program id of the program to upgrade.
+    #[clap(long)]
+    program_address: Pubkey,
+
+    /// The address that holds the new program data.
+    #[clap(long)]
+    buffer_address: Pubkey,
+
+    /// Account that will receive leftover funds from the buffer account.
+    #[clap(long)]
+    spill_address: Pubkey,
+}
+
+#[derive(Clap, Debug)]
+struct ShowMultisigOpts {
+    /// The multisig account to display.
+    #[clap(long)]
+    multisig_address: Pubkey,
+}
+
+#[derive(Clap, Debug)]
+struct ShowTransactionOpts {
+    /// The transaction to display.
+    #[clap(long)]
+    transaction_address: Pubkey,
+}
+
+#[derive(Clap, Debug)]
+struct ApproveOpts {
+    /// The multisig account whose owners should vote for this proposal.
+    // TODO: Can be omitted, we can obtain it from the transaction account.
+    #[clap(long)]
+    multisig_address: Pubkey,
+
+    /// The transaction to approve.
+    #[clap(long)]
+    transaction_address: Pubkey,
+}
+
+/// Resolve ~/.config/solana/id.json.
+fn get_default_keypair_path() -> PathBuf {
+    let home = std::env::var("HOME").expect("Expected $HOME to be set.");
+    let mut path = PathBuf::from(home);
+    path.push(".config/solana/id.json");
+    path
+}
+
+fn main() {
+    let opts = Opts::parse();
+    let payer_keypair_path = match opts.keypair_path {
+        Some(path) => path,
+        None => get_default_keypair_path(),
+    };
+    let payer = read_keypair_file(&payer_keypair_path)
+        .expect(&format!("Failed to read key pair from {:?}.", payer_keypair_path));
+
+    let client = Client::new_with_options(
+        Cluster::Localnet,
```

Comment:
Shouldn't this be configurable as a CLI Option?
like `--cluster localnet` or `--cluster mainnet`

Reply:
Yes, it should. Added in 05aef806abf068146ec675c57e3cc60e25ac85d1. Clap makes this surprisingly easy! (Thanks to the [`FromStr` impl in `anchor-client`](https://docs.rs/anchor-client/0.4.5/anchor_client/enum.Cluster.html#impl-FromStr) mostly).